packages/backend-common/api-report.md

```diff
@@ -110,6 +110,19 @@ export class AzureUrlReader implements UrlReader {
   toString(): string;
 }
 
+// @public
+export type BackstageEvent<
```

Comment:
I wonder if there's really any value in subscribing to a topic about _the entirety of Backstage_. I feel like the catalog is an entire product unto itself, which may affect the event topic design. Ping @pjungermann here of course, does that match how you think about proper granularity of these things?

My gut feeling tells me that we might choose to remove `BackstageEvent` and `BackstageEventPayload`, and instead add

```ts
export type CatalogEvent<
  TPayload extends CatalogEventPayload = CatalogEventPayload
> = EventParams<TPayload> & {
  topic: 'backstage.catalog',
};

export type CatalogEventPayload = {
  type: string;
};
```

to `@backstage/plugin-catalog-node`.

Then we could have more specific event subtypes there as well if we wanted to. If you want to use originating entity ref as a type-invariant routing parameter, then sure you could add that too but it's a bit of a large decision so if there's no strict need for it now, might want to hold off for a bit and see what patterns emerge as we add more types.

On that topic (ha!), would it even have been compelling to put that type in `@backstage/plugin-catalog-common` so that it can be used from the frontend as well when we have subscriptions there in the future? Which would mean that at least some of the types in this file could then be living in a new `packages/events` named `@backstage/events` - or possibly in `@backstage/plugin-events-common`, but haven't considered yet whether that might introduce any weird circular dependencies.

Reply:
I like these ideas! I removed `BackstageEvent` in favor of `CatalogEvent` in catalog-common in https://github.com/backstage/backstage/pull/17174/commits/66298c08baa1d99da2e328224762404af83a2405.

I'm unsure about an `events` package; some of the types like `EventBroker` don't seem sensible for the frontend to know about, so would it then just be `EventParams` that's common? Or just expose all the types to keep them together even if it might not make sense for the frontend?